director/director_advertise.go

```diff
@@ -136,12 +137,36 @@ func registerDirectorAd(appCtx context.Context, egrp *errgroup.Group, ctx *gin.C
 	}
 	directorAd := fAd.DirectorAd
 
+	// Reset the expiration times if we detect significant skew between the claimed time sent
+	// and the current time.
+	//
+	// That is, if the ad claims to have been sent at noon and expiring at 12:15 but the server
+	// received it at 13:00, then assume the expiration time is 13:15 (since the original lifetime
+	// was set to 15 minutes).
```

Comment:
Why is this the correct behavior? My assumption would have been we'd ignore the ad, since it'd appear to be pre-expired.

Reply:
This is a bit of historical experience coming from HTCondor.  If the client updates a timestamp on an ad as it is sending it and the server receives an ad with a large skew, then we assume that the client had a bad clock.  I wanted to plant this seed (the other option is to simply have a TTL and move it forward based on the server-side) so we can detect and report the clock skews in the future.